tests/tst_indexfilter.qml

```diff
@@ -0,0 +1,71 @@
+import QtQuick 2.0
+import SortFilterProxyModel 0.2
+import QtQml.Models 2.2
+import QtTest 1.1
+
+Item {
+    property list<IndexFilter> filters: [
+        IndexFilter {
+            property string tag: "basicUsage"
+            property int expectedModelCount: 3
+            property var expectedValues: [3, 1, 2];
+            minimumIndex: 1; maximumIndex: 3
+        },
+        IndexFilter {
+            property string tag: "outOfBounds"
+            property int expectedModelCount: 0
+            minimumIndex: 3; maximumIndex: 1
+        },
+        IndexFilter {
+            property string tag: "0to0Inverted"
+            property int expectedModelCount: 4
+            property var expectedValues: [3,1,2,4]
+            minimumIndex: 0; maximumIndex: 0; inverted: true
+        },
+        IndexFilter {
+            property string tag: "0to0" // bug / issue #15
+            property int expectedModelCount: 1
+            property var expectedValues: [5]
+            minimumIndex: 0; maximumIndex: 0
+        },
+        IndexFilter {
+            property string tag: "basicUsageInverted"
+            property int expectedModelCount: 2
+            property var expectedValues: [5,4]
+            minimumIndex: 1; maximumIndex: 3; inverted: true
+       }
+    ]
+
+    SortFilterProxyModel {
+        id: testModel
+        property ListModel myModel: ListModel {
```

Comment:
I believe it would be clearer to create the `ListModel` outside and reference it in the `SortFilterProxyModel` by its id.
This way we don't have to create another property in the proxy model.

BTW I have a lead on the crash when creating the `ListModel` inline in the `sourceModel` property, it should be easily fixable.